File: estrutura_de_dados/uri_1068.py
class Pilha():
    def __init__(self):
        self.stack = []

    def push(self, char):
        self.stack.append(char)

    def pop(self):
        if(self.stack != []):
            self.stack.pop()
            return 0
        else:
            return -1

    def empty(self):
        if self.stack == []:
            return 1
        else:
            return 0

n = int(input())

for i in range(n):
    expressao = input()

    aux = 0

    pilha = Pilha()

    for letra in expressao:
        if letra == '(':
            pilha.push(letra)
        elif letra == ')':
            aux = pilha.pop()
            if aux == -1:
                break
    
    if aux == 0 and pilha.empty():
        print('correct')
    else:
        print('incorrect')

    

File: estrutura_de_dados/uri_1068.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int pilha, n, i, j;
    i = 0;

    scanf("%d", &n);

    for(; i < n; i++)
    {
        pilha = j = 0;

        char expressao[1000];

        scanf("%s", expressao);

        int aux = 0;
        while(expressao[j] != '\0')
        {
            if(expressao[j] == '('){
                pilha++;
            }else if(expressao[j] == ')'){
                pilha--;
            }

            if(pilha < 0){
                aux = -1;
                printf("incorrect\n");
                break;
            }

            j++;
        }

        if(pilha == 0 && aux == 0){
            printf("correct\n");
        }

    }

    return 0;
}